File: countBattleships/countBattleships.go
package main

func countBattleships(board [][]byte) int {
	count := 0

	for i := range board {
		for j := range board[0] {
			if board[i][j] == 'X' && (j+1 == len(board[i]) || board[i][j+1] != 'X') && (i+1 == len(board) || board[i+1][j] != 'X') {
				count++
			}
		}
	}

	return count
}

File: Leetcode/countBattleships/countBattleships.js
module.exports = countBattleships = board => {
  let count = 0;
  for (let i = 0; i < board.length; i++) {
    for (let j = 0; j < board[0].length; j++) {
      if (board[i][j] === 'X' && board[i][j+1] !== 'X' && (!board[i+1] || board[i+1][j] !== 'X')) {
        count++
        }
    }
  }
  return count
};